File: push/src/main/java/com/xiaomi/push/service/XMPushServiceStateSupport.java
package com.xiaomi.push.service;

import com.xiaomi.channel.commonutils.network.Network;

final class XMPushServiceStateSupport {
    private XMPushServiceStateSupport() {
    }

    static boolean isPushDisabled(XMPushService service) {
        try {
            Class clsLoadClass = com.xiaomi.channel.commonutils.android.SystemUtils.loadClass(service, "miui.os.Build");
            java.lang.reflect.Field field = clsLoadClass.getField("IS_CM_CUSTOMIZATION_TEST");
            java.lang.reflect.Field field2 = clsLoadClass.getField("IS_CU_CUSTOMIZATION_TEST");
            java.lang.reflect.Field field3 = clsLoadClass.getField("IS_CT_CUSTOMIZATION_TEST");
            return field.getBoolean(null) || field2.getBoolean(null) || field3.getBoolean(null);
        } catch (Throwable th) {
            return false;
        }
    }

    static boolean shouldReconnect(XMPushService service) {
        return Network.hasNetwork(service) && PushClientsManager.getInstance().getActiveClientCount() > 0 && !service.isPushDisabled() && service.isPushEnabled() && !service.isSuperPowerModeEnable() && !service.isExtremePowerSaveMode();
    }
}

File: legacy-runtime/src/main/java/com/xiaomi/push/service/XMPushServiceStateSupport.kt
package com.xiaomi.push.service

import com.xiaomi.channel.commonutils.network.Network

class XMPushServiceStateSupport private constructor() {
    companion object {
        @JvmStatic
        fun isPushDisabled(service: XMPushService): Boolean {
            return try {
                val buildClass = com.xiaomi.channel.commonutils.android.SystemUtils.loadClass(service, "miui.os.Build")
                val isCmCustomizationTest = buildClass.getField("IS_CM_CUSTOMIZATION_TEST")
                val isCuCustomizationTest = buildClass.getField("IS_CU_CUSTOMIZATION_TEST")
                val isCtCustomizationTest = buildClass.getField("IS_CT_CUSTOMIZATION_TEST")
                isCmCustomizationTest.getBoolean(null) ||
                    isCuCustomizationTest.getBoolean(null) ||
                    isCtCustomizationTest.getBoolean(null)
            } catch (_: Throwable) {
                false
            }
        }

        @JvmStatic
        fun shouldReconnect(service: XMPushService): Boolean {
            return Network.hasNetwork(service) &&
                PushClientsManager.getInstance().getActiveClientCount() > 0 &&
                !service.isPushDisabled() &&
                service.isPushEnabled() &&
                !service.isSuperPowerModeEnable() &&
                !service.isExtremePowerSaveMode()
        }
    }
}
